src/sentry/replays/usecases/ingest/dom_index.py

```diff
@@ -0,0 +1,152 @@
+
+from django.conf import settings
+
+from sentry.utils import json, kafka_config, metrics
+from sentry.utils.pubsub import KafkaPublisher
+
+EVENT_LIMIT = 20
+
+replay_publisher: Optional[KafkaPublisher] = None
+
+ReplayActionsEventPayloadClick = TypedDict(
+    "ReplayActionsEventPayloadClick",
+    {
+        "node_id": int,
+        "tag": str,
+        "id": str,
+        "class": List[str],
+        "role": str,
+        "aria_label": str,
+        "alt": str,
+        "testid": str,
+        "title": str,
+        "text": str,
+        "timestamp": int,
+        "event_hash": str,
+    },
+)
+
+
+class ReplayActionsEventPayload(TypedDict):
+    type: Literal["replay_actions"]
+    replay_id: str
+    clicks: List[ReplayActionsEventPayloadClick]
+
+
+class ReplayActionsEvent(TypedDict):
+    type: Literal["replay_event"]
+    start_time: float
+    replay_id: str
+    project_id: int
+    retention_days: int
+    payload: List[bytes]
+
+
+def parse_and_emit_replay_actions(
+    project_id: int,
+    replay_id: str,
+    retention_days: int,
+    segment_data: List[Dict[str, Any]],
+) -> None:
+    with metrics.timer("replays.usecases.ingest.dom_index.parse_and_emit_replay_actions"):
+        message = parse_replay_actions(project_id, replay_id, retention_days, segment_data)
+        if message is not None:
+            publisher = _initialize_publisher()
+            publisher.publish("ingest-replay-events", json.dumps(message))
+
+
+def parse_replay_actions(
+    project_id: int,
+    replay_id: str,
+    retention_days: int,
+    segment_data: List[Dict[str, Any]],
+) -> Optional[ReplayActionsEvent]:
+    """Parse RRWeb payload to ReplayActionsEvent."""
+    actions = get_user_actions(replay_id, segment_data)
+    if len(actions) == 0:
+        return None
+
+    payload = create_replay_actions_payload(replay_id, actions)
+    return create_replay_actions_event(replay_id, project_id, retention_days, payload)
+
+
+def create_replay_actions_event(
+    replay_id: str,
+    project_id: int,
+    retention_days: int,
+    payload: ReplayActionsEventPayload,
+) -> ReplayActionsEvent:
+    return {
+        "type": "replay_event",
+        "start_time": time.time(),
+        "replay_id": replay_id,
+        "project_id": project_id,
+        "retention_days": retention_days,
+        "payload": list(json.dumps(payload).encode()),  # type: ignore
+    }
+
+
+def create_replay_actions_payload(
+    replay_id: str,
+    clicks: List[ReplayActionsEventPayloadClick],
+) -> ReplayActionsEventPayload:
+    return {
+        "type": "replay_actions",
+        "replay_id": replay_id,
+        "clicks": clicks,
+    }
+
+
+def get_user_actions(
+    replay_id: str,
+    events: List[Dict[str, Any]],
+) -> List[ReplayActionsEventPayloadClick]:
+    """Return a list of ReplayActionsEventPayloadClick types."""
+    result: List[ReplayActionsEventPayloadClick] = []
+    for event in events[:EVENT_LIMIT]:
```

Comment:
don't we want this to be the number of click events, not the total number of events in the payload? I think this wouldn't work very well as there are usually a lot in one segment

Reply:
Oof.  Thanks for catching that :D